### DISCUSSION TITLE
Not able to download RAW files

### DISCUSSION THREAD

**Discussion by broadsword0007D** [community/community#69838] [Actions]
### Select Topic Area

Bug

### Body

I am not able to download anything from GITHUB, I have tried downloading the raw files for my use but no action is working when I click the download button. Can anyone Please help me with this issue

**Comment by thevindu-w** [comment]
May be you can try with a different browser.
Is there any error shown?

**Reply by broadsword0007D** [reply]
I have tried, no error is shown just no action is there when I click the download icon. I have tried it in Mozilla too. The problem persists

**Reply by thevindu-w** [reply]
Is that file too large?

**Reply by broadsword0007D** [reply]
no. I am not able to download file of any size from anywhere on github

**Reply by thevindu-w** [reply]
Is this link working for you? [https://raw.githubusercontent.com/broadsword0007D/VIT-AI-ML-Aug-2023/main/README.md](https://raw.githubusercontent.com/broadsword0007D/VIT-AI-ML-Aug-2023/main/README.md)
If it's working (showing the content of the README.md file or downloading it), try this also [https://github.com/broadsword0007D/VIT-AI-ML-Aug-2023/raw/main/README.md](https://github.com/broadsword0007D/VIT-AI-ML-Aug-2023/raw/main/README.md)
Or is it only the download buttons that (probably) use some JavaScript that redirects to the download not working?

**Reply by broadsword0007D** [reply]
both the links are not working

**Reply by thevindu-w** [reply]
Can you check the network log in the browser dev tools when visiting those links? What is the status code (if present) or does that show any error?
Those links give me a page containing the content of the README.md like this. (it is the expected behavior)
![image](https://github.com/community/community/assets/91218789/48b086f3-c1f5-4031-a06b-b9d62603c049)

**Reply by broadsword0007D** [reply]
![image](https://github.com/community/community/assets/99869338/3b840936-4d1e-4516-95e6-f3368b75ab02)
This is the error the site is not loading

**Reply by broadsword0007D** [reply]
I don't know how to check network log in the browser

**Reply by thevindu-w** [reply]
But can you visit other pages in GitHub? For example, [https://github.com/broadsword0007D/VIT-AI-ML-Aug-2023/](https://github.com/broadsword0007D/VIT-AI-ML-Aug-2023/)

**Reply by broadsword0007D** [reply]
![image](https://github.com/community/community/assets/99869338/fa15ac51-90ef-4844-8e5c-f91383e9b309)

**Reply by broadsword0007D** [reply]
yeah I can visit the pages

**Reply by broadsword0007D** [reply]
![image](https://github.com/community/community/assets/99869338/078b696b-2c8d-486d-b715-b75c19a816ee)

**Reply by thevindu-w** [reply]
> I don't know how to check network log in the browser

Chrome: [https://developer.chrome.com/docs/devtools/open/](https://developer.chrome.com/docs/devtools/open/)
Firefox: [https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/index.html](https://firefox-source-docs.mozilla.org/devtools-user/network_monitor/index.html)

**Reply by thevindu-w** [reply]
Can you open the not working URL while network monitor is open?
I get this when opening that URL with status 200 (That means OK). You may get something different.
![image](https://github.com/community/community/assets/91218789/89223fa2-48be-42e2-8e0a-4a538237d93f)

**Reply by broadsword0007D** [reply]
![image](https://github.com/community/community/assets/99869338/ec99159d-4c2f-4be3-a45b-88be5de0b117)
this is the network monitor, right?
This is what I am getting in the network monitor

**Reply by thevindu-w** [reply]
Yes that's the network monitor. But I think you are applying a filter (Em3u8). Try removing that.
Also you need to reload the page while the network monitor is open.

**Reply by broadsword0007D** [reply]
![image](https://github.com/community/community/assets/99869338/96f2014c-edd3-4070-86d9-3dee9f417a96)

**Reply by thevindu-w** [reply]
I can see you don't get any response from `raw.githubusercontent.com`. May be because they have temporarily blocked your IP or your request don't reach the raw.githubusercontent.com server.
Try to ping it.
`ping raw.githubusercontent.com`
Run it in a command prompt / terminal

**Reply by broadsword0007D** [reply]
how check that and solve it?

**Reply by broadsword0007D** [reply]
![image](https://github.com/community/community/assets/99869338/02c4b5d7-9b2d-46c6-a29d-759dea13d324)

**Reply by thevindu-w** [reply]
Ok, probably your requests are not going to the server. But DNS is working. Are you sure that your ISP or some entity between you and the internet is not blocking that domain? If you are in an office/school/university or a similar network, the network admins can also block certain domains.
You can check with a different internet connection if you have one. Or try with a VPN or a proxy.

**Reply by broadsword0007D** [reply]
https://github.com/orgs/community/discussions/42655 
I got this, I have the same connection. Maybe the connection provider is blocking it but i don't know why

**Reply by thevindu-w** [reply]
May be. That discussion is also about the same problem of raw.githubusercontent.com being not accessible. Please try the solutions in that thread. Sometimes they may work.
Most of the solutions there are about changing the DNS.

**Reply by broadsword0007D** [reply]
Yeah, Thanks for helping out. Really man thanks a lot

**Reply by broadsword0007D** [reply]
I started using VPN and its working

**Comment by MRezaSarrafi** [comment]
Hello

**Comment by redaezziani** [comment]
hi

**Reply by broadsword0007D** [reply]
hello sir

**Comment by goyalyashpal** [comment]
i am facing same issue from several months. i posted my input at: https://github.com/orgs/community/discussions/77343#discussioncomment-7938843

thanks a lot @thevindu-w for your thorough diagnosis at https://github.com/orgs/community/discussions/69838#discussioncomment-7286748

i am on my personal connection, and i don't use any vpn. is there any other cure for this problem?

```console
$ ping raw.githubusercontent.com

Pinging raw.githubusercontent.com [2405:200:1607:2820:41::36] with 32 bytes of data:
Request timed out.
Request timed out.
Request timed out.
Request timed out.

Ping statistics for 2405:200:1607:2820:41::36:
    Packets: Sent = 4, Received = 0, Lost = 4 (100% loss),


```

**Reply by thevindu-w** [reply]
This implies you get the same ipv6 address 2405:200:1607:2820:41::36 as some others got in [github.com/orgs/community/discussions/69838#discussioncomment-7286759](https://github.com/orgs/community/discussions/69838#discussioncomment-7286759). But I think this IP address is no longer correct. Did you try a different DNS server like Google (8.8.8.8) or Cloudflair (1.1.1.1) to get the correct IP address? This thread [github.com/orgs/community/discussions/42655](https://github.com/orgs/community/discussions/42655) suggests that and some have got their issue solved in that thread.

Here are some IP addresses (v4 and v6) I got from 1.1.1.1 and 8.8.8.8
```
$ nslookup raw.githubusercontent.com 1.1.1.1
Server:		1.1.1.1
Address:	1.1.1.1#53

Non-authoritative answer:
Name:	raw.githubusercontent.com
Address: 185.199.111.133
Name:	raw.githubusercontent.com
Address: 185.199.108.133
Name:	raw.githubusercontent.com
Address: 185.199.110.133
Name:	raw.githubusercontent.com
Address: 185.199.109.133
Name:	raw.githubusercontent.com
Address: 2606:50c0:8002::154
Name:	raw.githubusercontent.com
Address: 2606:50c0:8000::154
Name:	raw.githubusercontent.com
Address: 2606:50c0:8003::154
Name:	raw.githubusercontent.com
Address: 2606:50c0:8001::154

$ nslookup raw.githubusercontent.com 8.8.8.8
Server:		8.8.8.8
Address:	8.8.8.8#53

Non-authoritative answer:
Name:	raw.githubusercontent.com
Address: 185.199.110.133
Name:	raw.githubusercontent.com
Address: 185.199.108.133
Name:	raw.githubusercontent.com
Address: 185.199.111.133
Name:	raw.githubusercontent.com
Address: 185.199.109.133
Name:	raw.githubusercontent.com
Address: 2606:50c0:8000::154
Name:	raw.githubusercontent.com
Address: 2606:50c0:8001::154
Name:	raw.githubusercontent.com
Address: 2606:50c0:8002::154
Name:	raw.githubusercontent.com
Address: 2606:50c0:8003::154
```

Try to ping some of those IP addresses (instead of the domain name) like,
```
ping 2606:50c0:8002::154
```
If ping those IP address works, then it confirms that your default DNS has the problem.

If DNS was the problem, you can switch to one of those DNS servers permanently (you may refer to the following links if needed).
Windows: [www.howtogeek.com/786911/how-to-change-your-dns-server-on-windows-10/](https://www.howtogeek.com/786911/how-to-change-your-dns-server-on-windows-10/)
Mac: [support.apple.com/en-lk/guide/mac-help/mh14127/mac](https://support.apple.com/en-lk/guide/mac-help/mh14127/mac)
Linux: [unix.stackexchange.com/a/494325](https://unix.stackexchange.com/a/494325)

**Reply by goyalyashpal** [reply]
> This thread [github.com/orgs/community/discussions/42655](https://github.com/orgs/community/discussions/42655) suggests tha

yep, i read that thread too after reading [author's comment](https://github.com/orgs/community/discussions/69838#discussioncomment-7286782). and while the author of that discussion has done a fantastic job, but the discussion there didn't felt much reliable. as many varying suggestions as many ppl chimed in there. i kinda got overwhelmed from there and din't do anything.

> If ping those IP address works, then it confirms that your default DNS has the problem.

<details><summary>yeah it worked 😃 </summary>

<br/>

```console
$ nslookup raw.github.com 1.1.1.1
Server:  one.one.one.one
Address:  1.1.1.1

Non-authoritative answer:
Name:    raw.github.com
Addresses:  2606:50c0:8002::154
          2606:50c0:8003::154
          2606:50c0:8000::154
          2606:50c0:8001::154
          185.199.108.133
          185.199.109.133
          185.199.111.133
          185.199.110.133


$ which nslookup
/c/Windows/system32/nslookup

$ nslookup raw.github.com 8.8.8.8 2>/dev/null | head -n2 # same as 1^4 but ipv6 sorted lol
Server:  dns.google
Address:  8.8.8.8

$ ping 2606:50c0:8000::154

Pinging 2606:50c0:8000::154 with 32 bytes of data:
Reply from 2606:50c0:8000::154: time=56ms 
Reply from 2606:50c0:8000::154: time=59ms 
Reply from 2606:50c0:8000::154: time=57ms 
Reply from 2606:50c0:8000::154: time=61ms 

Ping statistics for 2606:50c0:8000::154:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 56ms, Maximum = 61ms, Average = 58ms

```

</details>

> Windows: [www.howtogeek.com/786911/how-to-change-your-dns-server-on-windows-10/](https://www.howtogeek.com/786911/how-to-change-your-dns-server-on-windows-10/)

thanks a lottt 😇 
github repos are referred from soooo many developer places, that hald the web was feeling broken to me, now hopefully it will be solved 😇 

_edit:_ yeah, the above article starts with this very first line. 
~_so, one question: dns is provided by ISP by default?_~


-----
looking up a bit further... i wondery why or how does `nslookup` gets correct address when `server` ain't specified

```console
$ nslookup //?
Usage:
   nslookup [-opt ...]             # interactive mode using default server
   nslookup [-opt ...] - server    # interactive mode using 'server'
   nslookup [-opt ...] host        # just look up 'host' using default server
   nslookup [-opt ...] host server # just look up 'host' using 'server'

$ nslookup raw.github.com 2>/dev/null | head -n6
Server:  UnKnown
Address:  1**.1**.**.**

Name:    raw.github.com
Addresses:  2606:50c0:8002::154
          2606:50c0:8000::154

```

**Reply by thevindu-w** [reply]
> i wondery why or how does nslookup gets correct address when server ain't specified

It uses the default DNS configured in ths OS

**Reply by goyalyashpal** [reply]
this repo: https://github.com/blamy/vscode-chat-builder showed gif correctly - probably 'cz it uses https://github.com/assets ... which exposes these via https://private-user-images.githubusercontent.com/

https://github.com/BLamy/vscode-chat-builder/blob/416a45cb4420e7d801a53fa4cbc8acd6c67edaba/README.md?plain=1#L3

**Comment by whitesnakeftw** [comment]
Same problem here on Windows 10, only happens with Chromium-based browsers though (Chrome 121 and Edge), Firefox works fine. On Chrome I get this on my console:

`Access to fetch at 'https://raw.githubusercontent.com/pathtofile' (redirected from 'https://github.com/pathtofile') from origin 'https://github.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status.`

If I add `--disable-web-security` to Chrome's starting flags, it works, as suggested here: https://stackoverflow.com/questions/77716854/im-getting-this-error-response-to-preflight-request-doesnt-pass-access-contro

**Comment by zhongchengyong** [comment]
Same problem for me in 2025.  
I found a workaroud to download the file based on chrome: press 'F12' to open the DevTools, then click the 'download' button, then you will see there is download request in the 'Network' view, after that, select the request link and copy the html link right side to browser, the download should be start.

**Reply by JanPlayer2** [reply]
I have the same issue, but your workaround saved me, thx!
